common/core/web/tools/sentry-manager/src/index.ts

```diff
@@ -0,0 +1,137 @@
+  /**
+   * Controls whether or not the generated Sentry event is logged to the console (true)
+   * or sent to our Sentry server (false).
+   */
+  let DEBUG = false;
+
+  type Options = typeof KeymanSentryManager.DEFAULT_OPTIONS;
+
+  export class KeymanSentryManager {
+    keymanPlatform: string;
+
+    static STANDARD_ALIASABLE_FILES = {
+      'keymanweb.js':    'keymanweb.js',
+      'kmwuibutton.js':  'kmwuibutton.js',
+      'kmwuifloat.js':   'kmwuifloat.js',
+      'kmwuitoggle.js':  'kmwuitoggle.js',
+      'kmwuitoolbar.js': 'kmwuitoolbar.js'
+      // Also add entries for the naming system used by Android and iOS - and map them to the EMBEDDED upload, not the std 'native' one.
+    }
+
+    static DEFAULT_OPTIONS = {
+      hostPlatform: "native-web"
+    }
+
+    constructor(options: Options = KeymanSentryManager.DEFAULT_OPTIONS) {
+      this.keymanPlatform = options.hostPlatform;
+    }
+
+    // If we've recognized one of our source files, 
+    aliasFilename(filename: string): string|null {
+      if(!this.mayAlias(filename)) {
+        return null;
+      }
+
+      filename = KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+
+      switch(location.protocol) {
+        case 'http:':
+          return 'http://' + location.host + '/' + filename;
+        case 'file:':
+          return 'file:///' + filename;
+        default:
+          return null;
+      }
+    }
+
+    mayAlias(filename: string): boolean {
+      return !!KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+    }
+
+    // Filters all expected but unnecessary path prefixes, affixes, and suffixes reported by Sentry from our products.
+    // This allows us to mask all different sorts of installations with a single uploaded path.
+    pathFilter(event: any) {
+      // Get the underlying JS error.
+      let exception = event.exception;
+
+      // Iterate through all wrapped exceptions.
+      for(let e of exception.values) {
+        for(let frame of e.stacktrace.frames) {
+          let URL = frame.filename as string;
+          let filename: string = '';
+          try {
+            filename = URL.substr(URL.lastIndexOf('/')+1);
+            if(this.mayAlias(filename)) {
+              // Make the file look like it's at the host's root so that the Sentry server will recognize it
+              // regardless of actual filepath position.
+              frame.filename = this.aliasFilename(filename) || frame.filename;
+            }
+          } catch {
+            // not sure what to do yet, but... don't make more errors in the error handler.  That's a bad idea.
+          }
+        }
+      }
+
+      return event;
+    }
+
+    attachEventMetadata(event: any) {
+      // Ensure that the 'extra' object exists.  (May not exist for synthetic/custom Errors.)
+      event.extra = event.extra || {};
+      event.extra.keymanState = window['keyman']['getDebugInfo']();
+      event.extra.keymanHostPlatform = this.keymanPlatform;
+      return event;
```

Comment:
![Sentry 'extra' result](https://user-images.githubusercontent.com/25213402/81779164-596b8e80-951e-11ea-9cc3-0eb34dbef53e.png)

This screenshot was taken before the `keymanState` metadata field was added, but it shows up similarly.

Reply:
For something a bit more current and practical - https://sentry.keyman.com/share/issue/20515266ca584804bd5af29a4ee2b031/:

![Screen Shot 2020-05-13 at 1 45 26 PM](https://user-images.githubusercontent.com/25213402/81780106-25916880-9520-11ea-9557-03bd41ce3308.png)

This is generated from the newly-thrown `Error` during `setActiveKeyboard`'s `Promise` rejection.  By default, unhandled Promise rejections are remarkably uninformative; this pattern gives us better-quality data... like _which_ `Promise` was rejected.